04/app.js

```diff
@@ -0,0 +1,39 @@
+const min = 10;
+const max = 200;
+const randomArray = createArray(min, max);
+const largest = getLargest(randomArray);
+const avg = getAvg(largest);
+
+console.log("losowa tablica: ",randomArray);
+console.log("10 największych liczb: ",largest);
+console.log("średnia 10 najwiękzych liczb wynosi: ",avg);
+
+function createArray(min, max) {
+    const tab = [];
+    for(let i = 0; i < 20; i++) {
+        tab.push(Math.floor(Math.random()*(max - min) + min + 1));
+    }
+    // console.log(tab);
+    return tab;
+}
+
+function getLargest(arr) {
+    arr.sort(function(a, b) {
+        return a - b;
+    });
+    return arr.slice(arr.length/2, arr.length);
```

Comment:
Tutaj w zadaniu jest napisany o 10 największych. To rozwiązanie działa, ale czy jeśli podamy np. na wejściu tablicę z 30 elementami to otrzymamy 10 największych? Nie, raczej staramy się pisać elastyczny kod, ale nie możemy przesadzać. Nazwa funkcji nie wskazuje, że pobieramy połowę elementów, a w zadaniu jest mowa o dokładnie 10 dlatego powinniśmy się trzymać tego zalecenia. Tak jakbyśmy mieli zadanie w pracy. Potem w dokuemntacji jest napsiane, że mamy 10 elementów, a tak na prawdę mamy połowę z przekazanej tablicy. Subtelna różnica, ale potem może wiele namieszać :) ufff wybacz ten przydługawy wywód :D